internal/bminventory/inventory.go

```diff
@@ -144,12 +145,12 @@ type OCPClusterAPI interface {
 
 //go:generate mockgen --build_flags=--mod=mod -package bminventory -destination mock_installer_internal.go . InstallerInternals
 type InstallerInternals interface {
-	RegisterClusterInternal(ctx context.Context, kubeKey *types.NamespacedName, params installer.V2RegisterClusterParams) (*common.Cluster, error)
+	RegisterClusterInternal(ctx context.Context, kubeKey *types.NamespacedName, mirrorRegistryConfiguration *v1beta1.MirrorRegistryConfiguration, params installer.V2RegisterClusterParams) (*common.Cluster, error)
```

Comment:
Yeah, One of the definitions in this task is to not affect the REST API. Updating the `ClusterCreateParams`  can only be done with adding it to the swagger definition which will affect the REST API.

Reply:
I don't like that implementation also, from my perspective there are few solution for passing the `mirrorRegistryConfiguration`:
 - Change the internal API (as done here) and pass `mirrorRegistryConfiguration` if needed
 - Change `ClusterCreateParams` and include `mirrorRegistryConfiguration` which affect the REST API
 - Create a new structs that holds kubeapi parameters (e.g. `kubeKey`, `mirrorRegistryConfiguration` ), this struct will replace `kubeKey` and will prevent us from adding new parameters to the function declaration in the future. The main problem with this method is that it will require us to refactor the code and it a bit out of this PR scope

Reply:
You basically right, but the moment we expose the mirror registry configuration on swagger we will need to support some logic on the REST side (even if this is just to raise an error when the user supplying mirror registry configuration on non kube api mode).
@avishayt WDYT?